### ISSUE TITLE
Support for User Assigned Identity in azurerm_stream_analytics_job

### ISSUE THREAD

**Issue by Jezevecek** [hashicorp/terraform-provider-azurerm#19897]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently Terraform supports only System Assigned Identity, but usage of User Managed Identity is already available through Azure Portal.

### New or Affected Resource(s)/Data Source(s)

azurerm_stream_analytics_job

### Potential Terraform Configuration

```hcl
resource "azurerm_stream_analytics_job" "bc_template" {

  ..

  identity {
    type          = "UserAssigned"
    identity_id   = azurerm_user_assigned_identity.xxx.id
  }
}
```

**Comment by jiaweitao001** [comment]
Hi @Jezevecek , thanks for opening this issue. This is a feature enhancement of `azurerm_stream_analytics_job` resource, we will put it in our stack and work on it, but I cannot give an exact timeline now. I'll keep you posted on this thread. Thank you for your understanding!

**Comment by baselsalam** [comment]
@Jezevecek please help upvote this [identical request ](https://github.com/hashicorp/terraform-provider-azurerm/issues/18787) from Dec 2022 , which is blocked on [this request in "Azure SDK for Go"](https://github.com/Azure/azure-sdk-for-go/issues/19407)

**Comment by tombuildsstuff** [comment]
Taking a look through here this appears to be a duplicate of https://github.com/hashicorp/terraform-provider-azurerm/issues/18787 as @baselsalam has mentioned - as such I'm going to consolidate this into that issue - would you mind subscribing to that issue for updates?

Thanks!

**Comment by ryantuffney-shell** [comment]
Sure no problem.
